antarest/study/business/area_management.py

```diff
@@ -318,12 +318,21 @@ def remove_layer(self, study: RawStudy, layer_id: str) -> None:
 
     def create_area(self, study: Study, area_creation_info: AreaCreationDTO) -> AreaInfoDTO:
         file_study = self.storage_service.get_storage(study).get_raw(study)
+
+        # check if area already exists
+        existing_area_ids = set(file_study.config.areas)
```

Comment:
You can remove `existing_area_ids` set because checking directly in `file_study.config.areas` achieves the same without extra steps.
```python
if area_id in file_study.config.areas:
        raise DuplicateAreaName(area_creation_info.name)
```
The rest looks good to me 👍🏼